src/Kanvas/ImportersRequests/Actions/ImporterFromMapperAction.php

```diff
@@ -0,0 +1,74 @@
+<?php
+declare(strict_types=1);
+
+namespace Kanvas\ImportersRequests\Actions;
+
+use Kanvas\ImportersRequests\Models\ImporterRequest;
+use Kanvas\MappersImportersTemplates\Models\MapperImporterTemplate;
+use Kanvas\Filesystem\Models\Filesystem;
+use League\Csv\Reader;
+use Illuminate\Support\Facades\Storage;
+use Kanvas\Filesystem\Services\FilesystemServices;
+use Kanvas\Apps\Models\Apps;
+
+class ImporterFromMapperAction
+{
+    
+    public function __construct(
+        private ImporterRequest $importerRequest,
+        private MapperImporterTemplate $mapperImporterTemplate
+    ) {
+    }
+
+    public function execute(): void
+    {
+        
+        $path = $this->getFilePath($this->importerRequest->filesystem);
+        $reader = Reader::createFromPath($path, 'r');
+        $reader->setHeaderOffset(0);
+        $records = $reader->getRecords();
+        foreach ($records as $record) {
+            $data = $this->mapper($this->mapperImporterTemplate->mapper, $record);
+            $systemModuleImportDto = $this->mapperImporterTemplate->systemModules->importer_job::dispatchSync(
```

Comment:
@FredPeal why send each row to the importer and not the full batch?

Reply:
if one product fail, the next no